src/test/java/test/DynamicGraphTest.java

```diff
@@ -67,20 +67,18 @@ public void test8() {
     dg.addEdge(0, b1, x, y);
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
-    dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
```

Comment:
Keep the `RUNNING` steps but you can update their checks. 
Then add the `FINISHED` steps and their checks.
It will be more complete, isn't it?